File: javascript/src/get-client-credentials.js
import config from './config.js';
import { Moneyhub } from '@mft/moneyhub-api-client';
import yargs from 'yargs';
import { hideBin } from 'yargs/helpers';

const DEFAULT_SCOPES = 'accounts:read transactions:read:all';

const args = yargs(hideBin(process.argv))
  .option('userId', {
    alias: 'u',
    describe: 'userId to get token for',
  })
  .option('scope', {
    alias: 's',
    describe: 'scopes to request',
    default: DEFAULT_SCOPES,
  }).argv;

const main = async () => {
  const moneyhub = await Moneyhub(config);
  const url = await moneyhub.getClientCredentialTokens({
    scope: args.scope,
    sub: args.userId,
  });
  console.log(url);
};

main();

File: python/src/get_client_credentials.py
from config import *
import utils
import argparse

parser = argparse.ArgumentParser(description='Get client credentials token from Moneyhub')
parser.add_argument("-s", "--scope", default="user:read", help="The scope value you want a token for")
parser.add_argument("-u", "--userId", help="The userId you want the token to be scoped to")
args = parser.parse_args()

token = utils.get_client_credentials_token(config, args.scope, args.userId)

print(token)